docs/csharp/language-reference/keywords/volatile.md

```diff
@@ -9,42 +9,46 @@ helpviewer_keywords:
+ The `volatile` modifier is usually used for a field that is accessed by multiple threads without using the [lock](lock-statement.md) statement to serialize access.  
   
  The `volatile` keyword can be applied to fields of these types:  
   
--   Reference types.  
-  
--   Pointer types (in an unsafe context). Note that although the pointer itself can be volatile, the object that it points to cannot. In other words, you cannot declare a "pointer to volatile."  
+- Reference types.  
   
--   Types such as sbyte, byte, short, ushort, int, uint, char, float, and bool.  
+- Pointer types (in an unsafe context). Note that although the pointer itself can be volatile, the object that it points to cannot. In other words, you cannot declare a "pointer to volatile."  
   
--   An enum type with one of the following base types: byte, sbyte, short, ushort, int, or uint.  
+- Simple types such as `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `char`, `float`, and `bool`.  
   
--   Generic type parameters known to be reference types.  
+- An `enum` type with one of the following base types: `byte`, `sbyte`, `short`, `ushort`, `int`, or `uint`.  
   
--   <xref:System.IntPtr> and <xref:System.UIntPtr>.  
+- Generic type parameters known to be reference types. 
   
- The volatile keyword can only be applied to fields of a class or struct. Local variables cannot be declared `volatile`.  
+- <xref:System.IntPtr> and <xref:System.UIntPtr>.  
   
-## Example  
- The following example shows how to declare a public field variable as `volatile`.  
+The volatile keyword can only be applied to fields of a `class` or `struct`. Local variables cannot be declared `volatile`.
   
- [!code-csharp[csrefKeywordsModifiers#24](../../../csharp/language-reference/keywords/codesnippet/CSharp/volatile_1.cs)]  
-  
-## Example  
- The following example demonstrates how an auxiliary or worker thread can be created and used to perform processing in parallel with that of the primary thread. For background information about multithreading, see [Managed Threading](../../../standard/threading/index.md) and [Threading (C#)](../../programming-guide/concepts/threading/index.md).  
+## Example
+
+The following example shows how to declare a public field variable as `volatile`.  
   
- [!code-csharp[csProgGuideThreading#1](../../../csharp/language-reference/keywords/codesnippet/CSharp/volatile_2.cs)]  
+[!code-csharp[declareVolatile](~/samples/snippets/csharp/language-reference/keywords/volatile/Program.cs#Declaration)]
+
+The following example demonstrates how an auxiliary or worker thread can be created and used to perform processing in parallel with that of the primary thread. For background information about multithreading, see [Managed Threading](../../../standard/threading/index.md) and [Threading (C#)](../../programming-guide/concepts/threading/index.md).  
```

Comment:
PR #8164 redirects the C# guide article to the .NET guide article. So the end of the sentence ` and [Threading (C#)](../../programming-guide/concepts/threading/index.md)` can be removed.